### PR TITLE
feat: space ingest context enrichment & URI tracking

### PR DESCRIPTION
**Pull request by valentinyanakiev**
## Summary
- Prepends parent callout title + truncated description to each contribution (post, whiteboard, link) so chunked content retains hierarchical context for better RAG retrieval
- Propagates entity URLs from the Alkemio GraphQL API through `DocumentMetadata` → `StoreStep` → vector store, enabling clickable source links in expert responses
- Adds `url` to all GraphQL profile queries in the space tree reader

## Files Changed
- `core/domain/ingest_pipeline.py` — added `uri: str | None` field to `DocumentMetadata`
- `core/domain/pipeline/steps.py` — conditional `uri` inclusion in stored chunk metadata
- `plugins/ingest_space/space_reader.py` — GQL `url` fetching, callout context enrichment, URI propagation
- `specs/022-space-ingest-context-uri/` — full SDD artifact set

## Test plan
- [x] 12 new unit tests (callout context enrichment, URI propagation, StoreStep URI)
- [x] 3 existing tests fixed (updated `_process_space` signature and callout structure)
- [ ] Verify ingested space content in ChromaDB includes callout context prefix
- [ ] Verify stored metadata contains `uri` fields

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * URI tracking through the ingestion pipeline for source attribution and clickable links
  * Enriched space contributions with parent callout context to preserve hierarchy during processing
  * Content deduplication using normalized text hashing

* **Documentation**
  * Added specification, plan, quickstart, research, tasks, and checklist docs for the feature

* **Tests**
  * New unit tests validating URI propagation, context enrichment, truncation/edge cases, and storage behavior

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -260,6 +261,23 @@ async def _run(config: BaseConfig) -> None:
...
+        if gql_endpoint and admin_email:
+            deps["graphql_client"] = GraphQLClient(
+                graphql_endpoint=gql_endpoint,
+                kratos_public_url=kratos_url,
+                email=admin_email,
+                password=admin_password,
+            )
+            logger.info("GraphQL client configured: %s", gql_endpoint)
+        else:
+            logger.warning("GraphQL client not configured — missing API_ENDPOINT_PRIVATE_GRAPHQL or AUTH_ADMIN_EMAIL")
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply main.py]: Fixed — the guard now requires all three: `gql_endpoint and admin_email and admin_password`. The warning message has also been updated to mention `AUTH_ADMIN_PASSWORD`.

File: plugins/ingest_space/space_reader.py [hunk 2]
```diff
@@ -2,66 +2,93 @@
...
+    # Normalise whitespace.
+    text = _HORIZONTAL_WS_RE.sub(" ", text)
+    text = _EXCESS_NEWLINES_RE.sub("\n\n", text)
+    return text.strip()
+
+
+def _content_key(content: str) -> str:
+    """Whitespace-insensitive hash used for deduplication."""
+    normalised = re.sub(r"\s+", " ", content).strip().lower()
+    return hashlib.sha256(normalised.encode("utf-8")).hexdigest()
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply plugins/ingest_space/space_reader.py:47]: Declining — the content-based dedup is intentional. Identical content across entities means redundant embeddings in the vector store that waste storage and dilute retrieval quality. The `document_id` of the first-seen copy is preserved, which is sufficient for provenance. Content hash dedup downstream (StoreStep) also operates on content alone, so this is consistent with the pipeline's overall dedup strategy.

File: specs/022-space-ingest-context-uri/spec.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# Feature Specification: Space Ingest Context Enrichment & URI Tracking
+
+**Feature Branch**: `022-space-ingest-context-uri`
+**Created**: 2026-04-17
+**Status**: Implemented
+**Input**: Retrospec from code changes
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply specs/022-space-ingest-context-uri/spec.md:6]: Declining — "Retrospec" is our internal shorthand for "retroactive specification" (a spec generated from existing code changes). It is intentional project terminology, not a typo.
